executor/builder.go

```diff
@@ -73,11 +73,12 @@ var (
 // executorBuilder builds an Executor from a Plan.
 // The InfoSchema must not change during execution.
 type executorBuilder struct {
-	ctx        sessionctx.Context
-	is         infoschema.InfoSchema
-	snapshotTS uint64 // The consistent snapshot timestamp for the executor to read data.
-	err        error  // err is set when there is error happened during Executor building process.
-	hasLock    bool
+	ctx              sessionctx.Context
+	is               infoschema.InfoSchema
+	snapshotTS       uint64 // The consistent snapshot timestamp for the executor to read data.
+	snapshotTSCached bool
```

Comment:
Still need it?

Reply:
Yes. The reason why it's needed is in `getSnapshotTS`.

Reply:
I think it won't happen.

Reply:
I'm not sure. I think  it's better to keep the additional check before we are sure that this case never happens.

Reply:
What do you think @lzmhhh123 ?

Reply:
I think we still need this. For some SQLs with a limit statement, although it will be interrupted when required rows are satified. The operator will be canceled step by step(some internal goroutines). It may still under execution. Then the new operators will call `Open` function by `index join`.

Reply:
Is this case covered by any test?

Reply:
And by the way, does IndexLookupJoin has the same problem? I found that it didn't force cache the snapshotTS